### PR TITLE
[DOCS] Add retriever examples, semantic reranking step-by-step guide

### PR DESCRIPTION
**Pull request by leemthompo**
## ✨ [URL preview](https://elasticsearch_bk_111793.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/retriever.html)

Updates documentation on search retrievers and semantic reranking. Major changes include:

- Modification of example queries and descriptions in the "standard retriever" section to illustrate how to use basic and more complex boolean queries.
- Introduction of explanatory notes for query elements, improving documentation readability and understanding.
- Inclusion of a new "hybrid search" example that combines lexical and vector searches using RRF
- Enhancement of the "text similarity reranker retriever" section 
- An extended example of setting up semantic reranking with a Hugging Face model using Eland
- Clarify supported 3rd party similarity models

### CODE REVIEW

File: docs/reference/search/retriever.asciidoc [hunk 1]
```diff
@@ -257,27 +371,117 @@ Sets a minimum threshold score for including documents in the re-ranked results.
...
+                     "text":"landmark in Paris"
+                  }
+               }
+            }
+         },
+         "field":"text",
+         "inference_id":"my-cohere-rerank-model",
+         "inference_text":"Most famous landmark in Paris",
+         "rank_window_size":100,
+         "min_score":0.5
```
Comment (demjened) [inline docs/reference/search/retriever.asciidoc:400]: Nit: whitespace
```suggestion
   "retriever": {
      "text_similarity_reranker": {
         "retriever": {
            "standard": {
               "query": {
                  "match_phrase": {
                     "text": "landmark in Paris"
                  }
               }
            }
         },
         "field": "text",
         "inference_id": "my-cohere-rerank-model",
         "inference_text": "Most famous landmark in Paris",
         "rank_window_size": 100,
         "min_score": 0.5
```

File: docs/reference/search/retriever.asciidoc [hunk 2]
```diff
@@ -257,27 +371,117 @@ Sets a minimum threshold score for including documents in the re-ranked results.
...
++
+. Upload the model to {es} using Eland. This example assumes you have an Elastic Cloud deployment and an API key. Refer to the https://www.elastic.co/guide/en/elasticsearch/client/eland/current/machine-learning.html#ml-nlp-pytorch-auth[Eland documentation] for more authentication options.
++
+[source,sh]
+----
+eland_import_hub_model \
+  --cloud-id $CLOUD_ID \
+  --es-api-key $ES_API_KEY \
+  --hub-model-id cross-encoder/ms-marco-MiniLM-L-6-v2 \
+  --task-type c \
```
Comment (demjened) [inline docs/reference/search/retriever.asciidoc:436]: Does this work like this? I think it has to be an exact value from an enum, in this case `text_similarity`.

**Review by demjened — APPROVED** [review]
LGTM, few minor comments.

File: docs/reference/search/retriever.asciidoc (hunk 2 above)
Comment (leemthompo) [reply docs/reference/search/retriever.asciidoc:436]: that's a fat-fingered typo :)

File: docs/reference/search/retriever.asciidoc [hunk 3]
```diff
@@ -180,34 +215,112 @@ An RRF retriever is a compound retriever. Child retrievers may not use
...
+{
+  "retriever": {
+    "rrf": {
+      "retrievers": [
+        {
+          "standard": {
+            "query": {
+              "sparse_vector": {
+                "field": "plot_embedding",
+                "inference_id": ".elser_model_2",
```
[the remark this answers is not in the document]
Comment (leemthompo) [reply docs/reference/search/retriever.asciidoc:286]: 💯

File: docs/reference/search/retriever.asciidoc (hunk 2 above)
Comment (leemthompo) [inline docs/reference/search/retriever.asciidoc:436]:
```suggestion
  --task-type text_similarity \
```

**Pull request merged**